### DISCUSSION TITLE
About renaming/aliasing --shamefully-hoist

### DISCUSSION THREAD

**Discussion by mauricioszabo** [pnpm/pnpm#3670] [Ideas]
Hi! First of all, GREAT WORK on this library - it's literally making my SSD consume 50gb less than it usually was!

So... I've been working with ClojureScript, and I can _only work_ with `pnpm` if I install things with `--shamefully-hoist`. As I'm not a native English speaker, it wasn't clear that's the feature I wanted, so I had to carefully read the help and try what it feels right, and the last one I tried was this one (because of the "shamefully", probably).

So... how about aliasing to something, like `--include-all-subdeps` or `--make-subdeps-accessible`,  or something like this?

The reason, _I believe_ that I can only with with ClojureScript with this options is probably because, on that language, there's a different way to resolve dependencies that's in, most cases, compatible with Javascript (but not always, and it may have small differences in some situations). It as also not clear, in the beginning, why `pnpm` wasn't working as both `npm` and `yarn` work correctly :)

**Comment by zkochan** [comment]
We can add instructions to the docs about how to use pnpm with ClojureScript

**Reply by mauricioszabo** [reply]
That will also help a lot! Thanks :)
